SYCL/ESIMD/api/functional/operators/operator_assignment.hpp

```diff
@@ -0,0 +1,96 @@
+//===-- operator_assignment.hpp - Functions for tests on simd assignment
+//      operators. --------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides functions for tests on simd assignment operators.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+// The test proxy is used to verify the move assignment was called actually.
+#define __ESIMD_ENABLE_TEST_PROXY
+
+#include "../common.hpp"
+#include "../shared_element.hpp"
+
```

Comment:
Do we really need to distinguish the place to include `shared_element` and `shared_vector`? That's the same level of abstraction, so looks like we might just include shared_element into the same header as the shared_vector

Reply:
`shared_element` not needed in all tests, so this inclusion will be useless in such cases.
Therefore I suggest do not include in in SYCL\ESIMD\api\functional\ctors\common.hpp